### DISCUSSION TITLE
Resolving non public dependency in constructor

### DISCUSSION THREAD

**Discussion by DaiJyoBu** [dadhi/DryIoc#413] [Q&A] [answered]
Hello,
here's the thing I'd like to do:
First part looks like that:

    public interface Interface1
    {
    }

    internal interface Interface1internal
    {
    }

    internal class Class1 : Interface1, Interface1internal
    {
    }

It registers and resolves ok - as "Interface1".
Second part:

    

    public interface Interface2
    {
    }

    internal class Class2 : Interface2
    {
        public Class2(Interface1internal ii)
        {
        }
    }

Does not resolve - "Class2"'s constructor is public but it requires internal interface as the argument.
Is there a way to make it working without replacing "Interface1internal" or turning it into public?
Thanks in advance!

**Comment by dadhi** [comment, marked as answer]
@DaiJyoBu How did you register with the internal interface?

**Reply by DaiJyoBu** [reply]
I've been using RegisterMany, is this method not the one I am to use when registering internal interfaces?

**Reply by dadhi** [reply]
@DaiJyoBu

`RegisterMany` has an optional parameter, something like `nonPublic: true` to register internal service types.

**Reply by DaiJyoBu** [reply]
Tried out using `RegisterMany` like that - worked! Thank you very much.

**Reply by dadhi** [reply]
Cool!
